### PR TITLE
allow adding columns in query field

### PR DESCRIPTION
**Pull request by crissi**
## Summary
Allow adding columns in query closure. Fixes: https://github.com/rebing/graphql-laravel/issues/875

People might  be using addSelect() in their model relationship or have a need to ad an extra column to a query direcly in the custom query.

Type of change:
- [x] Bug fix (non-breaking change which fixes an issue)

Checklist:
- [x ] Existing tests have been adapted and/or new tests have been added
- [x] Add a CHANGELOG.md entry
- [x] Update the README.md
- [x] Code style has been fixed via `composer fix-style`

### CODE REVIEW

File: src/Support/SelectFields.php [hunk 1]
```diff
@@ -98,12 +98,12 @@ public static function getSelectableFieldsAndRelations(
...
         return function ($query) use ($with, $select, $customQuery, $requestedFields, $ctx): void {
+            $query->select($select);
+            $query->with($with);
+            
             if ($customQuery) {
                 $query = $customQuery($requestedFields['args'], $query, $ctx);
             }
-
-            $query->select($select);
-            $query->with($with);
```
Comment (mfn) [inline src/Support/SelectFields.php:106]: This is a BC break: previously, `$customQuery` could returned a very different/completely custom `$query` but now that's not possible anymore.

For the fullest flexibility, we would need both:
- a `$createCustomQuery` (i.e. the current functionality)
- a `$adaptQuery` (i..e. what you're proposing)

**Review by mfn — CHANGES_REQUESTED** [review]
See my inline feedback, this is a BC break.

I can see the argument for your use case, but then we break the others and there's no "right or wrong" here. We currently support the one case and afterwards only the other.

Basically for this to work, we would need to keep the current `'query'` and add another one, invoked afterwards.

Interestingly, now that your https://github.com/rebing/graphql-laravel/pull/799 is merged, you could extend `SelectFields`, override `getSelectableFieldsAndRelations` and make any logic you want…

What do you think? How do you want to proceed here? I can't merge "as is" as I don't see a good argument made for changing/dropping the _existing_ behavuour.

File: src/Support/SelectFields.php (hunk 1 above)
Comment (crissi) [reply src/Support/SelectFields.php:106]: Hmm,  I might have meant to use addSelect, let me modify that and see if that changes anything

**Comment by crissi** [comment]
Hi mfn. 
I opened again using, addSelect and not reordering the way with and select  works

**Review by mfn — COMMENTED** [review]
Looks more elegant, but technically it's still change of behaviour…

Test coverage LGTM and I also can't think of case were this actually would break.

Are we sure we've no concerns here? Or should be rather wait for a major version bump?

We would also need a good / explanatory changelog for that one, so ppl understand the impact of the change.

WDYT?

**Comment by crissi** [comment]
could add it to this example: https://github.com/rebing/graphql-laravel#type-relationship-query
so it also included addSelect( 

I personally dont think this can cause any errors.

**Review by mfn — COMMENTED** [review]
> could add it to this example

Ok, let's do this. Yes, please add it :)

**Comment by crissi** [comment]
> > could add it to this example
> 
> Ok, let's do this. Yes, please add it :)

added

**Review by mfn — APPROVED** [review]
Thank you!

**Pull request merged**